tools/plugins/com.liferay.ide.upgrade.plan.ui/src/com/liferay/ide/upgrade/plan/ui/internal/tasks/UpgradeTaskStepItem.java

```diff
@@ -149,34 +149,6 @@ protected IStatus run(IProgressMonitor progressMonitor) {
 
 		_disposables.add(() -> fillLabel.dispose());
 
-		Image taskStepRestartImage = UpgradePlanUIPlugin.getImage(UpgradePlanUIPlugin.TASK_STEP_RESTART_IMAGE);
```

Comment:
If this step has actions, then it still needs the restart link.  We only want to avoid this if there are no actions as Children.